### DISCUSSION TITLE
Godaddy - acme challenge does not appear

### DISCUSSION THREAD

**Discussion by siarheikuzmich** [polymind-inc/acmebot#398] [Q&A] [answered]
We use GoDaddy as DNS provider. In the dashboard of Acmebot we can see list of our DNS zones and choose any. Also we can issue certificates in required zone (it takes about several minutes, I don't know why), then we see this certs in our key vault and import them to our services, but we did not find any records of acme challenge (token) in GoDaddy. As a result I guess GoDaddy can not approve certs and our sites can not see the right chain, for now sites show us expired GoDaddy certs chain that we used before.
Are there any ways to know, maybe, token and make txt record manually or anything else? Because we tried to issue cert twice and in both cases record did not appear

**Comment by shibayan** [comment, marked as answer]
GoDaddy has a minimum TTL setting of 10 minutes, so it should take at least 10 minutes to issue a certificate.

The token required for certificate issuance will be automatically removed from the DNS after the process is complete. There is no way to obtain the token manually.

**Reply by siarheikuzmich** [reply]
> GoDaddy has a minimum TTL setting of 10 minutes, so it should take at least 10 minutes to issue a certificate.
Yea, issuance takes relatively a long time.

> The token required for certificate issuance will be automatically removed from the DNS after the process is complete. There is no way to obtain the token manually.

Should Acmebot make a record in GoDaddy when we issue certs? or maybe there is some cache, because our site shows godaddy expired certs, but we upload a letsencrypt cert for our application. Lets I show in screenshots:
![image](https://user-images.githubusercontent.com/92019053/142443770-6acb2414-56bb-4723-b790-fc82bec04494.png)
![image](https://user-images.githubusercontent.com/92019053/142444367-43ba5b88-3d2c-4cea-8304-9200ee0463d6.png)

I don't know how to resolve this.

**Reply by shibayan** [reply]
If the certificate is stored in the Key Vault, Acmebot is working correctly. From that point on, please get Microsoft support for further use.

**Reply by siarheikuzmich** [reply]
Thank you to support in this issue.

> If the certificate is stored in the Key Vault, Acmebot is working correctly. From that point on, please get Microsoft support for further use.

I am afraid that some of steps were completed not fully during auto-issuance. My head pain with next questions: How we can be sure and exclude our side of the issue? How can our dns provider trust to this cert? How trust process is going?

**Reply by shibayan** [reply]
Check if the certificate has been created in Key Vault or if there is an error in Application Insights.

This question does not appear to be related to Acmebot, so it will be closed unless it is clear that an error has occurred.

**Reply by siarheikuzmich** [reply]
Thank you for quick response.
I did not find any Errors in logs. I removed old cert from app gateway, app services and so on, anywhere, rebinded custom domains to new cert, also reconfigured all listeners and http settings. Troubleshoot tool returns that "we have new valid certificate with XXX... thumberprint, but pages return thumberprint XXX... of old cert". Looks like some of services returns old cert, I dont know which of service, should investigate.

**Reply by shibayan** [reply]
It has nothing to do with Acmebot, I suggest you contact Azure support.

**Reply by siarheikuzmich** [reply]
I found out that if I change DNS record part of my custom domains start working with new cert. Looks like infrastructure issues. Thank you for help!
